rust/connlib/snownet/src/allocation.rs

```diff
@@ -501,6 +478,13 @@ impl Allocation {
                     return true;
                 };
 
+                // If we refreshed with a lifetime of 0, we deleted our previous allocation.
+                // Make a new one.
+                if lifetime.lifetime().is_zero() {
+                    self.authenticate_and_queue(make_allocate_request(), None);
+                    return true;
+                }
+
```

Comment:
I think it makes it a bit harder to read that the `make_delete_allocation_request` creates a request with a lifetime of 0 but in the same function we call that helper we check for zero.

We should either inline that function above or use a helper for this.

Reply:
I don't quite follow what you mean. We make a normal allocate here, not a refresh request.